front/src/reducers/osrdconf/osrdConfCommon/__tests__/utils.ts

```diff
@@ -600,6 +602,224 @@ const testCommonConfReducers = (slice: OperationalStudiesConfSlice | StdcmConfSl
+      const store = createStore(slice, {
+        pathSteps,
+      });
+
+      store.dispatch(slice.actions.addViaV2(lemansVia));
+      const state = store.getState()[slice.name];
+      expect(state.pathSteps).toStrictEqual([brest, lemans, strasbourg]);
+    });
+
+    it('should correctly append a new via point when the existing via is closer to the origin', () => {
+      const pathSteps = [brest, rennes, strasbourg];
+      const store = createStore(slice, {
+        pathSteps,
+      });
+
+      store.dispatch(slice.actions.addViaV2(lemansVia));
+      const state = store.getState()[slice.name];
+      expect(state.pathSteps).toStrictEqual([brest, rennes, lemans, strasbourg]);
+    });
+
+    it('should insert a via between two existing ones based on distance from origin', () => {
+      const pathSteps = [brest, rennes, paris, strasbourg];
+      const store = createStore(slice, {
+        pathSteps,
+      });
+
+      store.dispatch(slice.actions.addViaV2(lemansVia));
+      const state = store.getState()[slice.name];
+      expect(state.pathSteps).toStrictEqual([brest, rennes, lemans, paris, strasbourg]);
+    });
+
+    it('should insert a via at the end of the route', () => {
+      const pathSteps = [brest, rennes, lemans, strasbourg];
+      const store = createStore(slice, {
+        pathSteps,
+      });
+
+      store.dispatch(slice.actions.addViaV2(parisVia));
+      const state = store.getState()[slice.name];
+      expect(state.pathSteps).toStrictEqual([brest, rennes, lemans, paris, strasbourg]);
+    });
+  });
+
+  it('should handle permuteViasV2', () => {
+    const pathSteps = testDataBuilder.buildPathSteps();
+    const [brest, rennes, lemans, paris, strasbourg] = pathSteps;
+
+    const store = createStore(slice, { pathSteps: [brest, rennes, lemans, paris, strasbourg] });
+
+    store.dispatch(slice.actions.permuteViasV2(pathSteps, 0, 2));
+    const state = store.getState()[slice.name];
+    expect(state.pathSteps).toStrictEqual([brest, lemans, paris, rennes, strasbourg]);
+  });
+
+  describe('should handle upsertViaFromSuggestedOP', () => {
+    // For this action pathfinding has already been made so we know
+    // all steps will have a positionOnPath
+    const pathStepsData = testDataBuilder
+      .buildPathSteps()
+      .map((step, i) => step && { ...step, positionOnPath: i * 100 });
+
+    const [brest, rennes, lemans, paris, strasbourg] = pathStepsData;
+
+    it('should insert a new via if it comes from the suggested vias modal', () => {
+      const pathSteps = [brest, rennes, paris, strasbourg];
+      const store = createStore(slice, {
+        pathSteps,
+      });
+
+      const newVia: SuggestedOP = {
+        prId: 'lemans',
+        track: '60ca8dda-6667-11e3-81ff-01f464e0362d',
+        offsetOnTrack: 426.443,
+        positionOnPath: 200,
+        uic: 396002,
+        coordinates: [47.99542250806296, 0.1918181738752042],
+      };
+
+      const insertedVia: PathStep = {
+        id: 'id1',
```

Comment:
How do you know that the id will be id1 ? Is it not generated by nextId ?

Reply:
It's the one generated by nextId(). I'll leave a comment.